src/main/java/eu/bbmri_eric/negotiator/service/DiscoverySynchronizationJobServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package eu.bbmri_eric.negotiator.service;
+
+import eu.bbmri_eric.negotiator.database.model.DiscoveryService;
+import eu.bbmri_eric.negotiator.database.model.DiscoveryServiceSynchronizationJob;
+import eu.bbmri_eric.negotiator.database.model.DiscoveryServiceSyncronizationJobStatus;
+import eu.bbmri_eric.negotiator.database.repository.DiscoveryServiceRepository;
+import eu.bbmri_eric.negotiator.database.repository.DiscoveryServiceSynchronizationJobRepository;
+import eu.bbmri_eric.negotiator.dto.syncjobservice.DiscoverySyncJobServiceCreateDTO;
+import eu.bbmri_eric.negotiator.dto.syncjobservice.DiscoverySyncJobServiceDTO;
+import eu.bbmri_eric.negotiator.publishers.DiscoveryServiceSynchronizationEventPublisher;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import lombok.extern.apachecommons.CommonsLog;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@CommonsLog
+@Service
+public class DiscoverySynchronizationJobServiceImpl implements DiscoverySynchronizationJobService {
+
+  @Autowired DiscoveryServiceRepository discoveryServiceRepository;
+
+  @Autowired DiscoveryServiceSynchronizationEventPublisher publisher;
+
+  @Autowired
+  DiscoveryServiceSynchronizationJobRepository discoveryServiceSynchronizationJobRepository;
+
+  @Override
+  public DiscoverySyncJobServiceDTO createSyncJob(
+      DiscoverySyncJobServiceCreateDTO discoverySyncJobServiceCreateDTO) {
+    String discoveryServiceName = discoverySyncJobServiceCreateDTO.getDiscoveryServiceName();
+    Optional<DiscoveryService> discoveryService =
+        discoveryServiceRepository.findByName(discoveryServiceName);
+    if (!discoveryService.isEmpty()) {
+      LocalDateTime creationDate = LocalDateTime.now();
+      LocalDateTime modifyDate = LocalDateTime.now();
+      DiscoveryServiceSynchronizationJob job =
+          DiscoveryServiceSynchronizationJob.builder()
```

Comment:
use mapper

Reply:
Fixed in 3a6688e83ae3a2214da370432b86f40523292504